src/libsyncengine/syncpal/virtualfilescleaner.h

```diff
@@ -32,7 +32,7 @@ class VirtualFilesCleaner {
                             bool (*vfsStatus)(int, const SyncPath &, bool &, bool &, bool &, int &),
                             bool (*vfsClearFileAttributes)(int, const SyncPath &));
 
-        VirtualFilesCleaner(const SyncPath &path);
+        VirtualFilesCleaner(const SyncPath &path, int syncDbId);
```

Comment:
As discussed off line, passing the `syncDbId` as a constructor argument makes sure the created object is properly associated to a synchronization. This can be important for instance if we want to use sync-specific logging within the methods of a `VirtualFilesCleaner` object that was created with this lightweight constructor. 

As you pointed, it is actually not enough to guarantee that the created object can be handled safely by the caller. This is why I created https://infomaniak.atlassian.net/browse/KDESKTOP-1238.